### PR TITLE
feat: `mongodb` driver

### PR DESCRIPTION
**Pull request by itpropro**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -622,6 +622,50 @@ const storage = createStorage({
 - `storage`: The name of the table to read from. It defaults to `storage`.
 - `boostCache`: Whether to enable cached queries: see [docs](https://planetscale.com/docs/concepts/query-caching-with-planetscale-boost#using-cached-queries-in-your-application).
 
+### `mongodb`
+
+⚠️ Due to it's dependency on the `mongodb` package. this driver is currently not compatible with edge workers like Cloudflare Workers or Vercel Edge Functions.
```
Comment (danielroe) [inline README.md:627]:
```suggestion
⚠️ Due to its dependency on the `mongodb` package. this driver is currently not compatible with edge workers like Cloudflare Workers or Vercel Edge Functions.
```

File: README.md [hunk 2]
```diff
@@ -622,6 +622,50 @@ const storage = createStorage({
...
+```js
+import { createStorage } from "unstorage";
+import mongodbDriver from "unstorage/drivers/mongodb";
+const storage = createStorage({
+    driver: mongodbDriver({
+      connectionString: "CONNECTION_STRING",
+      databaseName: "test",
+      collectionName: "test",
+    }),
+});
```
Comment (danielroe) [inline README.md:654]:
```suggestion
const storage = createStorage({
  driver: mongodbDriver({
    connectionString: "CONNECTION_STRING",
    databaseName: "test",
    collectionName: "test",
  }),
});
```

File: src/drivers/mongodb.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+export default defineDriver((opts: MongoDbOptions = {}) => {
+  const {
+    connectionString,
+    databaseName = "unstorage",
+    collectionName = "unstorage",
+  } = opts;
+  if (!connectionString)
+    throw new Error(
+      "MongoDb driver requires a connection string to be provided."
```
Comment (danielroe) [inline src/drivers/mongodb.ts:29]:
```suggestion
      "MongoDB driver requires a connection string to be provided."
```

File: test/drivers/mongodb.test.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        expect(() =>
+          driver({ databaseName: "test", collectionName: "test" })
+        ).toThrowError();
+      });
+      it("should have different dates when an entry was updated", async () => {
+        await ctx.storage.setItem("s1:a", "test_data");
+        await sleep(100);
+        await ctx.storage.setItem("s1:a", "updated_test_data");
+        const result = await ctx.storage.getMeta("s1:a");
+        console.log(result);
```
Comment (danielroe) [inline test/drivers/mongodb.test.ts:36]:
```suggestion
```

File: src/drivers/mongodb.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      return;
+    },
+    async removeItem(key) {
+      await getMongoDbClient().deleteOne({ key });
+    },
+    async getKeys() {
+      const documents = await getMongoDbClient().find().toArray();
+      const keys = [];
+      documents.map((document) => keys.push(document.key));
+      return keys;
```
Comment (danielroe) [inline src/drivers/mongodb.ts:67]: What about:

```suggestion
      return await getMongoDbClient()
        .find()
        .map((document) => document.key)
        .toArray();
```

**Review by danielroe — APPROVED** [review]
This is great! Just a few very minor comments.

File: src/drivers/mongodb.ts (hunk 5 above)
Comment (itpropro) [reply src/drivers/mongodb.ts:67]: Passes all tests, I think I thought it would return an iterator, but it returns a cursor so this is a much better way 👍

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Thanks!

**Pull request merged**